lxd/instance/instance_utils.go

```diff
@@ -866,6 +895,10 @@ func CreateInternal(s *state.State, args db.InstanceArgs, clearLogDir bool) (Ins
 		if err != nil {
 			return err
 		}
+		// The `proxy` devices might collide with existing forward networks
+		if err = validateProxyDevCollisions(ctx, proj, s.DB.Cluster, tx, devices); err != nil {
```

Comment:
This would only get called at instance creation time, but we need it to occur at instance NIC start type.
This should be part of NIC `checkAddressConflict` I think.

Reply:
Oh wait we are talking about network forwards and proxy devices, not NICs. Sorry multi-tasking...badly.
But we do need something similar in the proxy device to check for address conflicts from network forwards, and then something for checking network forwards don't conflict.